WordPress/WordPressTest/CommentService+LikesTests.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+import Nimble
+import OHHTTPStubs
+import XCTest
+
+@testable import WordPress
+
+final class CommentService_LikesTests: CoreDataTestCase {
+
+    private var commentService: CommentService!
+
+    override func setUp() {
+        super.setUp()
+        commentService = CommentService(coreDataStack: contextManager)
+    }
+
+    override func tearDown() {
+        HTTPStubs.removeAllStubs()
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func test_likeComment_givenSuccessfulAPICall_updateLikesCount() {
+        let post = ReaderPost(context: mainContext)
+        post.siteID = 1
+        post.postID = 2
+        let comment = Comment(context: mainContext)
+        comment.commentID = 3
+        comment.post = post
+        contextManager.saveContextAndWait(mainContext)
+
+        // Add a successful HTTP API call stub
+        stub(condition: isPath("/rest/v1.1/sites/1/comments/3/likes/new")) { _ in
+            HTTPStubsResponse(
+                jsonObject: [String: Any](),
+                statusCode: 200,
+                headers: nil
+            )
+        }
+
+        // Call the toggle like function and wait for it to complete
+        waitUntil { done in
+            self.commentService.toggleLikeStatus(for: comment, siteID: 1) {
```

Comment:
Have you considered instantiating `commentService` as a `let` in each test method? 

#### Pro

- Guaranteed to be released when the test finished (a bit of a straw man, though given reassigning in `setUp` does the same)
- I _think_ `XCTestCase`s stays in memory until the whole test suite finishes, method-level lifecycle means the service instance won't stick around
- No need to access via `self`

#### Cons

- Not DRY (some would say, [WET](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself#WET))

Reply:
Sounds good to me! I don't see any benefit in declaring an instance property for it.